guide/concepts/events/debugging.md

```diff
@@ -0,0 +1,121 @@
+Sometimes it can be difficult to understand exactly what's happening in ChatEngine. ChatEngine
+includes some handy utilities to make debugging and profiling easier.
+
+## Debug Mode
+
+This mode will output **every event** happening within ChatEngine system and it's
+payload. The event log tells a story about what's happening inside. Events are logged
+as they are triggered locally.
+
+You can turn on debug mode by supplying ```debug: true``` within the second
+configuration object in {@link ChatEngine#create}.
+
+```js
+ChatEngine = ChatEngineCore.create({
+    //...
+}, {
+    debug: true
+});
+```
+
+This will output every event that occurs within ChatEngine. Note that not
+every event is a network event.
+
+
+![](/debugging-debug.png)
+
+```sh
+debug: newListener $.ready
+debug: $.created.chat { chat: {} }
+```
+
+This is effectively the same as writing:
+
+```js
+ChatEngine.onAny((event, payload) => {
+   console.log(event, payload);
+});
+```
+
+Beware that this is very verbose and should not be enabled in production as it
```

Comment:
Just a preference thing, could we partial bold this sentence? To stress the importance?